pkg/collector/corechecks/sbom/processor.go

```diff
@@ -177,11 +177,23 @@ func (p *processor) processHostRefresh() {
 	}
 
 	if err := p.sbomScanner.Scan(scanRequest, p.hostScanOpts, ch); err != nil {
-		log.Errorf("Failed to generate SBOM for host: %s", err)
+		log.Errorf("Failed to trigger SBOM generation for host: %s", err)
+		close(ch)
```

Comment:
Generally speaking it's not mandatory to close channel, and it should usually be done by the sender (as sending to closed channel will panic, and sender is not aware).